server/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java

```diff
@@ -548,6 +548,10 @@ private void processNormalizerFactory(
             TokenizerFactory keywordTokenizerFactory,
             Map<String, TokenFilterFactory> tokenFilters,
             Map<String, CharFilterFactory> charFilters) {
+        if (keywordTokenizerFactory == null) {
+            throw new IllegalStateException("keyword tokenizer factory is null");
```

Comment:
nit: maybe IllegalArgumentException but it really doesn't matter that much here.

Reply:
Maybe the message should mention that this requires the `analysis-common` module?

Reply:
@cbuescher I like this keep throwing `IllegalStateException` if we detect this far in starting up that there is no keyword tokenizer then we have detected way to late that we need the analysis-common module and in a sense this is then a misconfiguration, not really a missing setting.

@nik9000 I will update the error message.